src/Maestro/SubscriptionActorService/LocalGit.cs

```diff
@@ -46,40 +45,52 @@ public async Task<string> GetPathToLocalGitAsync()
             if (!string.IsNullOrEmpty(_gitExecutable) && File.Exists(_gitExecutable))
             {
                 _logger.LogInformation($"Git executable found at {_gitExecutable}. Checking the installation.");
-                // We should also mke sure that the git executable that exists runs properly
+                // We should also make sure that the git executable that exists runs properly
                 try
                 {
                     LocalHelpers.CheckGitInstallation(_gitExecutable, _logger);
                     return _gitExecutable;
                 }
                 catch
-                { 
+                {
                     _logger.LogWarning($"Something went wrong with validating git executable at {_gitExecutable}. Downloading new version.");
                 }
             }
 
-            await _semaphoreSlim.WaitAsync();
+            // Since the collisions are already using paths from _tempFiles.GetFilePath (instance-specific) we'll try
+            // a deterministic modification of that for the name of the semaphore (using a raw path as the name throws)
+            string semaphoreName = _tempFiles.GetFilePath("git-download-semaphore").Replace('/', '-').Replace('\\', '-').Replace(':', '-');
+            Semaphore crossProcessSemaphore = new Semaphore(1, 1, semaphoreName);
+            crossProcessSemaphore.WaitOne(TimeSpan.FromMinutes(5));
+
+            string gitLocation = _configuration.GetValue<string>("GitDownloadLocation", null);
```

Comment:
I'd move all this stuff inside the `try`. As it is if gitLocation isn't a valid absolute uri the semaphore won't be released.

Reply:
done, try {} now is immediately before the WaitOne.